src/App.tsx

```diff
@@ -1,15 +1,45 @@
-import React from 'react';
-import './App.scss';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import './App.scss';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isFocusedQuery, setIsFocusedQuery] = useState(false);
+  const [highlatedPerson, setHighlatedPerson] = useState<Person>(
+    peopleFromServer[0]
+  );
+  const { name, born, died } = highlatedPerson;
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
in this function you should reset selected person

Reply:
Do not understand this point